### PR TITLE
ios_ntp module

### PR DESCRIPTION
**Pull request by lvrfrc87**
New ios_ntp module for Cisco IOS.  This is a working DRAFT version based on nxos_ntp plus other arguments. I have submitted this PR to ask support regarding 1 issue I have:

- the module is not idempotent. I have pretty much followed the structure of other module (ios_logging, ios_banner) but I cannot figure out what I am missing

### CODE REVIEW

File: lib/ansible/modules/network/ios/ios_ntp.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    server:
+        description:
+            - Network address of NTP server.
+    peer:
+        description:
+            - Network address of NTP peer.
+    source_int:
+        description:
+            - Source interface for NTP requests.  
+     state:
```
Comment (Qalthos) [inline lib/ansible/modules/network/ios/ios_ntp.py]:
```suggestion
    state:
```
I don't know if that's the only problem, but it's the one that's sticking out to me

**Review by Qalthos — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/ios/ios_ntp.py (hunk 1 above)
Comment (lvrfrc87) [reply lib/ansible/modules/network/ios/ios_ntp.py]: That should be just a doc string, right? :)

**Comment by lvrfrc87** [comment]
Module is now idempotent for absent and present status. Given the below playbook:

```
- ios_ntp:
    server: 8.8.9.9
    source_int: Loopback0
    acl: NTP_ACL
    logging: true
    key_id: 10
    auth_key: 15435A030726242723273C21181319000A
    auth: true
    state: present
```

```
changed: [uk-tempvpn8] => {
    "changed": true,
    "commands": [
        "ntp server 8.8.9.9",
        "ntp source Loopback0",
        "ntp access-group peer NTP_ACL",
        "ntp logging",
        "ntp authenticate",
        "ntp trusted-key 10",
        "ntp authentication-key 10 md5 15435A030726242723273C21181319000A 7"
[...]
```

```
ok: [uk-tempvpn8] => {
    "changed": false,
    "commands": [],
[...]
```

**Review by lvrfrc87 — COMMENTED** [review]
fixed

**Comment by lvrfrc87** [comment]
Notes:

https://github.com/ansible/ansible/pull/50145/files

` test/units/modules/network/ios`  ` test/integration/targets/ios_ntp`

**Comment by lvrfrc87** [comment]
Create a new PR with test files #50705

**Pull request closed without merging**